internal/grpc/services/gateway/storageprovider.go

```diff
@@ -1028,32 +1028,20 @@ func (s *svc) find(ctx context.Context, ref *provider.Reference) (provider.Provi
 	return client, p[0], err
 }
 
-// findSpacesProvider looks up the provider that is responsible for the given request
+// findSpacesProvider looks up the spaces provider that is responsible for the given request
 // It will return a client that the caller can use to make the call, as well as the ProviderInfo. It:
 // - contains the provider path, which is the mount point of the provider
 // - may contain a list of storage spaces with their id and space path
 func (s *svc) findSpacesProvider(ctx context.Context, ref *provider.Reference) (provider.SpacesAPIClient, *registry.ProviderInfo, error) {
-	p, err := s.findSpaces(ctx, ref)
-	if err != nil {
-		return nil, nil, err
-	}
-
-	client, err := s.getSpacesProviderClient(ctx, p[0])
-	return client, p[0], err
-}
-
-func (s *svc) findUnique(ctx context.Context, ref *provider.Reference) (provider.ProviderAPIClient, *registry.ProviderInfo, error) {
 	p, err := s.findSingleSpace(ctx, ref)
 	if err != nil {
 		return nil, nil, err
 	}
 
-	client, err := s.getStorageProviderClient(ctx, p[0])
+	client, err := s.getSpacesProviderClient(ctx, p[0])
 	return client, p[0], err
 }
 
-// FIXME findAndUnwrap currently just returns the first provider ... which may not be what is needed.
-// for the ListRecycle call we need an exact match, for Stat and List we need to query all related providers
```

Comment:
the gateway is no longer smart. it does not aggregate any directories. clients have to do that now.